Assets/_BForBoss/_Weapons/Scripts/Equipment/EquipmentBehaviour.cs

```diff
@@ -50,13 +55,15 @@ private void SetupWeapons()
             }
 
             _weapons[_currentWeaponIndex].ActivateWeapon = true;
+            _meleeBehaviour.Initialize(_meleeWeaponInputAction, () => _playerPivotTransform);
```

Comment:
I was thinking of injecting it into the `Initializer` of EquiptmentBehaviour.  But i'lll approve, and let you move it there if you want to.